src/propagation.cpp

```diff
@@ -521,11 +587,23 @@ ot::expected<std::unique_ptr<ot::SpanContext>> SpanContext::deserialize(
           } else if (has_prefix(key, baggage_prefix)) {
             baggage.emplace(std::string{std::begin(key) + baggage_prefix.size(), std::end(key)},
                             value);
+          } else if (equals_ignore_case(key, headers_impl.tags_header)) {
+            trace_tags = deserializeTags(value);
+            const auto found_tag = trace_tags.find(upstream_services_tag);
+            if (found_tag != trace_tags.end()) {
+              upstream_services = deserializeUpstreamServices(found_tag->second);
```

Comment:
What should happen when this throws?

Right now, we pretend that the header didn't exist.

Perhaps instead we might...

- forward the value verbatim
- revisit the idea of parsing the whole thing and just extract info we need
    - this would leverage the fact that we currently do not change existing sampling priority
- ...?